### ISSUE TITLE
[Documentation] Define basic style for landing page

### ISSUE THREAD

**Issue by pylapp** [Orange-OpenSource/ouds-ios#326]
# Context

Today our technical documentation is exposed through _GitHub Pages_ and available at [ios.unified-design-system.orange.com](https://ios.unified-design-system.orange.com/).
The content of the **gh-pages** is composed by assets of _doccarchive_ folders exported using _Xcode_.
These elements are then processed so as to extract the assets we want, forge the final "web site" and create the landing page (https://github.com/Orange-OpenSource/ouds-ios/issues/168).
The current script making such operation is [this one](https://github.com/Orange-OpenSource/ouds-ios/blob/develop/docs_release/uploadWebDoc.sh). It copies and pastes HTML, CSS and JS files, and also merges some JSON.

These assets, picked from _doccarchives_ files are in fact generated through _documentation catalogs_ and DocC comments: thus the doc we have in our code is then reflected in a web fashion.

We export as many _doccarchives_ as modules defined in our _Swift Package_.

Today, no Apple tool using the _DocC_ documentation framework is able to output an HTML page pointing to all of these modules. Thus we use [this script](https://github.com/Orange-OpenSource/ouds-ios/blob/develop/docs_release/uploadWebDoc.sh) to forge an HTML file.

More details about the documentation toolchain avaialble [in this wiki page](https://github.com/Orange-OpenSource/ouds-ios/wiki/53-%E2%80%90-About-the-documentation).

# Issue

That page is ugly, but it does the job:
- it exposes hyperlinks pointing to all modules
- the library version is displayed so as to notify the user the version of the doumentation lib he / she is reading
- it refers also the GitHub project and the wiki
- it contains the build number of the documentation

But the page is ugly and not _Orange branded_. 

# Solution?

Use [Boosted](https://boosted.orange.com/) to customize the page to make it nicer, at least this [the ODS web site](https://orange-opensource.github.io/ods-ios/) 
The script generating this main page must be keept as simple as possible. We can consider exracting the HTML code inside a side file, but we won't add more assets not framework.

The page can be [found here](https://github.com/Orange-OpenSource/ouds-ios/blob/gh-pages/docs/index.html)

[Attached: HTML page source](https://github.com/user-attachments/files/17963220/index.html.zip)

**Comment by pylapp** [comment]
Maybe @louismaximepiton or @julien-deramond know some tips or starter kits or the use of _Boosted_.

@hannahiss I may remember you have a quite similar need in your side.

**Comment by louismaximepiton** [comment]
Maybe we can wait for a common design landing page for web documentation and we will backport it in here once we have it. On the stack, if we want at least an header or footer, I'm afraid that we will need to have Boosted as a dependency for now. But let's wait for a design.

**Comment by hannahiss** [comment]
Yes I asked for a design for our homepage at least, but there has been 0 discussion about it so I'm not sure at all the need has been understood... Today Frederic assigned the issue to me in Jira 😦 ??!!
Issue is there: https://portail.agir.orange.com/browse/OUDS-397 
For the header/footer it will probably stay as is for now

**Comment by pylapp** [comment]
ℹ️ FYI _blocking_ no guidelines nor skills to make a one page / one file doc

**Comment by B3nz01d** [comment]
@vprothais could you or someone in the team help clarify the design needs so that @Franco-Riccitelli can help provide the necessary elements?
cc: @MaximeTonnerre

**Comment by pylapp** [comment]
Keep in mind we don't want a pure website or a bunch of files to manage as we generate the index file by script and the more files we will have to manage the more boring and complex it will be.
If all resources can be referenced in the HTML page without additional files it could be nice.

**Comment by pylapp** [comment]
Quelques précisions :
- on ne peut customiser comme on veut la documentation générée par Apple ([donc chaque doc listée ic](https://ios.unified-design-system.orange.com/)i)
- le seul besoin (si c'en est réellement un) réside dans le changement de look de la page "chapeau"
- cette page est un [fichier HTML](https://github.com/Orange-OpenSource/ouds-ios/blob/gh-pages/docs/index.html) généré [par un script](https://github.com/Orange-OpenSource/ouds-ios/blob/develop/docs_release/uploadWebDoc.sh) avec quelques métadonnées sur le versioning
- tout est hébergé dans une [branche dédiée exposée via GitHub Pages](https://github.com/Orange-OpenSource/ouds-ios/tree/gh-pages)
- le site de la documentation est construit en recopiant des dizaines de milliers de fichiers HTML, CSS et JS généré depuis Xcode

Donc en soit il faudra juste se concentrer sur une page en pensant que comme elle est générée, le moins de fichiers on met à côté le mieux c'est.

Pour rappel, le choix de l'outil de génération de doc a été motivé par le fait de générer depuis Xcode du HTML depuis du markdown et du code Swift en reproduisant au mieux l'expérience développeur d'Apple et en réduisant aussi la charge qu’impliquerait d'avoir à gérer une doc à part comme sur ODS.

Le sujet de la doc a été abordé notamment ici : https://github.com/Orange-OpenSource/ouds-ios/issues/9, https://github.com/Orange-OpenSource/ouds-ios/issues/168, https://github.com/Orange-OpenSource/ouds-ios/issues/11

**Je propose donc que l'on nous fournisse un fichier HTML pour remplacer celui existant ([ici](https://github.com/Orange-OpenSource/ouds-ios/blob/gh-pages/docs/index.html)) avec l'ensemble des feuilles de styles et ressources qui vont bien en minimisant le plus possibles le recours à des fichiers à ajouter à côté pour garder les choses simples.

N'ayant pas les compétences en développement web, et ayant d'autres priorités, avis aux volontaires !**

**Comment by pylapp** [comment]
The tool in use to build and upload the online documentation has evolved.
We do not have a kind of "umbrella" page listing each module pages.
Eveyrything is now live at https://ios.unified-design-system.orange.com/documentation.
**Thus the topic to define a custom stylesheet is not relevant anymore as everything is generated.**

However, the discussion is not closed, and can be continued on dicussion https://github.com/Orange-OpenSource/ouds-ios/discussions/626.
A subthread has been opened here: https://github.com/Orange-OpenSource/ouds-ios/discussions/626#discussioncomment-12923677

Is a suitable solution has been defined, an issue can be created.
